addons/mail/models/mail_thread.py

```diff
@@ -267,10 +267,31 @@ def create(self, vals_list):
             threads_no_subtype = self.env[self._name]
             for thread in threads:
                 subtype = thread._creation_subtype()
-                if subtype:  # if we have a subtype, post message to notify users from _message_auto_subscribe
-                    thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
-                else:
+                if not subtype:
                     threads_no_subtype += thread
+                    continue
+                # if we have a subtype, post message to notify users from _message_auto_subscribe
+                relations = subtype._get_auto_subscription_subtypes(self._name)[4]
+                if relations:
+                    get_model = self.env["ir.model"]._get
+                    model_name = get_model(self._name).display_name
+                    parent_models = [
+                        f'"{thread[f_name].display_name}" ({get_model(p_model_name).display_name})'
+                        for p_model_name, f_names in relations.items()
+                        for f_name in f_names
+                        if thread[f_name]
+                    ]
+                    if len(parent_models) == 1:
+                        body = _('A new %(model_name)s has been created in the %(parent_model_name)s you follow.',
+                                 model_name=model_name, parent_model_name=parent_models[0])
+                    elif len(parent_models) > 1:
+                        body = _('A new %(model_name)s has been created in one of the %(parent_model_names)s you follow.',
+                                 model_name=model_name, parent_model_names=', '.join(parent_models))
+                    else:
+                        body = ''
+                else:
+                    body = ''
```

Comment:
you can put this at the beginning of the method are remove the else

Reply:
Moved just before the if (as we iterate on thread and the body is different for each one of them) and removed the 2 else.
(I should have misunderstood one of your previous comment because that change is actually a revert on the initial code)